GitCommands/Git/GitModule.cs

```diff
@@ -389,6 +389,47 @@ public static bool IsValidGitWorkingDir(string dir)
                    Directory.Exists(dirPath + "refs");
         }
 
+        /// <summary>
+        /// Asks git to resolve the given relativePath
+        /// git special folders are located in different directories depending on the kind of repo: submodule, worktree, main
+        /// See https://git-scm.com/docs/git-rev-parse#git-rev-parse---git-pathltpathgt
+        /// </summary>
+        /// <param name="relativePath"></param>
+        /// <returns></returns>
+        public string GetGitPath(string relativePath)
```

Comment:
I prefer to include `Git` in the method name. `GitModule` also operates on regular repository files, so I want to distinct these two cases. I renamed it to ResolveGitInternalPath.